src/console/MeasurementCollectionToFhir/Processor.cs

```diff
@@ -34,15 +38,51 @@ public class Processor : IEventConsumer
             _templateManager = templateManager;
             _measurementImportService = measurementImportService;
             _logger = logger;
+            _retryPolicy = CreateRetryPolicy(logger);
         }
 
         public async Task ConsumeAsync(IEnumerable<IEventMessage> events)
         {
             EnsureArg.IsNotNull(events);
             EnsureArg.IsNotNull(_templateDefinition);
```

Comment:
If the _templateDefinition is null, I think we will end up with the 100% cpu issue here. I think if you do something similar to what you did in the Normalize Processor, that would address this issue.

Reply:
The EnsureArg is called outside of the retryPolicy.ExecutAsync, so throwing an exception here shouldn't cause us to retry forever. 
I do, however, see the EnsureArg check being thrown inside of the retryPolicy.ExecuteAsync in the Normalization Processor. I'll adjust that

Reply:
@wi-y  Any particular reason why we check the validity of the `_templateDefinition` outside of the constructor? From the looks of it, If its null/empty there is never an opportunity to update it.